### PR TITLE
Sanitize Kubernetes Names

### LINKED ISSUE #151: Validate identifiers
- identifiers have a 63 character limit
- consistently replace under score characters `_` with dashes `-` (`_` -> `-`)

### PR DESCRIPTION
**Pull request by ckadner**
**Which issue is resolved by this Pull Request:** 
Resolves #151 

**Description of your changes:**
`sanitize_k8s_name` cleans and converts the names in the workflow according to [K8s name and label spec](https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#names)

**Environment tested:**
* Python Version (use `python --version`): `3.7`
* Tekton Version (use `tkn version`): `0.11`
* Kubernetes Version (use `kubectl version`): `0.15`
* OS (e.g. from `/etc/os-release`): `MacOS 10.15.3`

### CODE REVIEW

**Comment by ckadner** [comment]
@Tomcli -- when you get a chance, could you help review this PR? Thanks :-)

@Udiknedormin -- please let me know if this PR addresses the identifier issues when compiling and running your pipelines (#150, #151). Thank you.

File: sdk/python/kfp_tekton/compiler/_op_to_template.py [hunk 1]
```diff
@@ -490,9 +490,15 @@ def _op_to_template(op: BaseOp, enable_artifacts=False):
     if processed_op.pod_annotations or processed_op.pod_labels:
         template.setdefault('metadata', {})  # Tekton change, don't wipe out existing metadata
         if processed_op.pod_annotations:
-            template['metadata']['annotations'] = processed_op.pod_annotations
+            template['metadata']['annotations'] = {
+                sanitize_k8s_name(key, True, True, True, 253): value
```
Comment (Tomcli) [inline sdk/python/kfp_tekton/compiler/_op_to_template.py:494]: nit: There are some True flags for allowing dot/underscore/slash. Maybe is better to use [python keyword arguments](https://www.w3schools.com/python/gloss_python_function_keyword_arguments.asp) to explicitly show the special characters we are allowing here?

File: sdk/python/kfp_tekton/compiler/_op_to_template.py [hunk 2]
```diff
@@ -490,9 +490,15 @@ def _op_to_template(op: BaseOp, enable_artifacts=False):
...
+                for key, value in processed_op.pod_annotations.items()
+            }
         if processed_op.pod_labels:
-            template['metadata']['labels'] = processed_op.pod_labels
+            template['metadata']['labels'] = {
+                sanitize_k8s_name(key, True, True, True, 253): sanitize_k8s_name(value, True, True)
```
Comment (Tomcli) [inline sdk/python/kfp_tekton/compiler/_op_to_template.py:499]: nit: same as above.

File: sdk/python/kfp_tekton/compiler/_op_to_template.py (hunk 1 above)
Comment (ckadner) [reply sdk/python/kfp_tekton/compiler/_op_to_template.py:494]: I wanted to keep those code blocks as short as possible, but you are right, it is easier to read and prevents errors in the future if we use explicit parameter names

File: sdk/python/kfp_tekton/compiler/_op_to_template.py (hunk 2 above)
Comment (ckadner) [reply sdk/python/kfp_tekton/compiler/_op_to_template.py]: I pushed another commit. Now using keyword arguments where there were more than one `True`/`False` flags in a row

**Comment by ckadner** [comment]
Thanks @Tomcli

**Pull request merged**